### PR TITLE
Claim Reward Spec

### PR DESCRIPTION
**Pull request by rbharath**
I made some progress on getting this test running, but I run into problems when I try to have the applicant commit a vote. Here's the corresponding section of the TCR test:

https://github.com/skmgoldin/tcr/blob/master/test/registry/claimReward.js#L34

For some reason, the last line here causes the test to crash.

@robrobbins Any thoughts on obvious mistakes here?

### CODE REVIEW

**Comment by rbharath** [comment]
Ok, I think I finally get what's going on here. In order to add this spec correctly, we will need to add a new function `voterReward` that reveals the reward that voters earn to the higher order contract, which exposes the `voterReward` method of the underlying `Registry` contract. Once that's added this test will be functional. Will add this test in later today or tomorrow

File: spec/computable/registry/claim-reward.spec.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    parameterizer = new Parameterizer(owner)
+    const parameterizerAddress = await parameterizer.deploy(web3, { tokenAddress, votingAddress })
+    parameterizer.setProvider(provider)
+
+    registry = new Registry(owner)
+    const registryAddress = await registry.deploy(web3, { tokenAddress, votingAddress, parameterizerAddress, name: NAME })
+    registry.setProvider(provider)
+
+    //owner approves voting and reg to spend
+    // await erc20.approve(votingAddress, 1000000)
```
Comment (robrobbins) [inline spec/computable/registry/claim-reward.spec.ts]: remove the comments here , looks like voting is approved further down

File: spec/computable/registry/claim-reward.spec.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import Registry from '../../../src/contracts/registry'
+import { ParameterDefaults, NAME } from '../../../src/constants'
+import {
+  deployDll,
+  deployAttributeStore,
+  maybeParseInt,
+  eventReturnValues,
+} from '../../../src/helpers'
+
+const provider:any = ganache.provider(),
```
Comment (robrobbins) [inline spec/computable/registry/claim-reward.spec.ts]: this is the old http-provider style, now deprecated. you'll need to setup the newer style with a beforeAll and AfterAll to get the websocket-provider for web3.

File: spec/computable/registry/claim-reward.spec.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  it('should transfer the correct number of tokens once a challenge has been resolved', async () => {
+    const listBytes = stringToBytes(web3, 'listing.net'),
+      voterStartingBalance = maybeParseInt(await erc20.balanceOf(voter))
+
+    //// Apply
+    const tx1 = registry.apply(listBytes, ParameterDefaults.MIN_DEPOSIT, '', { from: applicant })
+    expect(tx1).toBeTruthy()
+
+    //// Challenge
+    const challID = eventReturnValues('_Challenge',
```
Comment (robrobbins) [inline spec/computable/registry/claim-reward.spec.ts]: eventReturnValues is from the older http style specs, there are websocket listeners now rike `onData`. you can find them in the other specs

File: src/contracts/registry.ts [hunk 4]
```diff
@@ -180,4 +194,12 @@ export default class extends Deployable {
...
   }
+
+ /**
+  * Return the voter's token reward for the given poll.
+  */
+  async voterReward(voter:string, challengeId:string, salt:Nos): Promise<Nos> {
+    const deployed = this.requireDeployed()
+    return await deployed.methods.voterReward(voter, challengeId, salt).call()
+  }
 }
```
Comment (robrobbins) [inline src/contracts/registry.ts]: good job here.

File: spec/computable/registry/claim-reward.spec.ts (hunk 1 above)
Comment (rbharath) [reply spec/computable/registry/claim-reward.spec.ts]: Done

File: spec/computable/registry/claim-reward.spec.ts (hunk 2 above)
Comment (rbharath) [reply spec/computable/registry/claim-reward.spec.ts]: Ok, think I got this figured out.

File: spec/computable/registry/claim-reward.spec.ts (hunk 3 above)
Comment (rbharath) [reply spec/computable/registry/claim-reward.spec.ts]: This is stymieing me a bit. Can we punt this fix to a different PR?

**Comment by rbharath** [comment]
@robrobbins Awesome, thanks for the edits. I think I get the idea of using emitters better now

**Comment by rbharath** [comment]
Coordinated with @robrobbins offline and this PR is now ready to merge in.

**Pull request merged**